lib/hocon/impl/parseable.rb

```diff
@@ -101,7 +101,11 @@ def parse
     self.class.force_parsed_to_object(parse_value(options))
   end
 
-  def parse_value(base_options)
+  def parse_value(base_options = nil)
+    if base_options.nil?
+      base_options = options
+    end
+
```

Comment:
Could you change the default value of this param to just be `options`?  So, `def parse_value(base_options = options)`.  It's not exactly the same, but maybe sufficient?